lib/error.g

```diff
@@ -212,33 +241,11 @@ BIND_GLOBAL("ErrorInner",
         if ErrorLevel = 0 then LEAVE_ALL_NAMESPACES(); fi;
         JUMP_TO_CATCH(0);
     fi;
-    PrintTo("*errout*", "Error, ");
-    for x in earlyMessage do
-        PrintTo("*errout*", x);
-    od;
-    if printThisStatement then 
-        if context <> GetBottomLVars() then
-            PrintTo("*errout*", " in\n  ");
-            PRINT_CURRENT_STATEMENT("*errout*", context);
-            PrintTo("*errout*", " called from \n");
-        else
-            PrintTo("*errout*", "\n");
-        fi;
-    else
-        location := CURRENT_STATEMENT_LOCATION(context);
-        if location <> fail then
-          PrintTo("*errout*", " at ", location[1], ":", location[2]);
-        fi;
-        PrintTo("*errout*", " called from\n");
-    fi;
-
     if SHOULD_QUIT_ON_BREAK() then
         FORCE_QUIT_GAP(1);
     fi;
 
-    if IsBound(OnBreak) and IsFunction(OnBreak) then
-        OnBreak();
-    fi;
```

Comment:
Huh, so `OnBreak` is now called *before*` the test for `SHOULD_QUIT_ON_BREAK`? That's quite a change in semantics, one which I am quite concerned about. I'll need to look at your whole PR with a bit more time to understand what is going on.
(It might help if the refactoring did not happen at the same time as the changes in behavior -- note that to me, "refactoring" means a change which does *not* change behavior (or at most changes in a very minimalistic way, which then however should be documented in the commit message)

Reply:
You're totally right. I'll make separate commits out of that. Note to myself: never mix refactoring and behaviour changes in a single commit.

I realize that I forgot to check for `SHOULD_QUIT_ON_BREAK()` in the if-clause that controls whether `OnBreak()` is called. That change in semantics was not on purpose.